modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/steps/StepsTable.scala

```diff
@@ -165,28 +169,36 @@ object StepsTable {
 
     // Columns for the table
     private def columns(p: Props, s: Size): List[Table.ColumnArg] = {
-      val (offsetVisible, exposureVisible, fpuVisible, filterVisible, objectSize) = s.width match {
-        case w if w < PhoneCut      => (false, false, false, false, SSize.Tiny)
-        case w if w < LargePhoneCut => (false, true, false, false, SSize.Small)
-        case _                      => (displayOffsets(p), true, true, true, SSize.Small)
+      val (offsetVisible, exposureVisible, disperserVisible, fpuVisible, filterVisible, objectSize) = s.width match {
+        case w if w < PhoneCut      => (false, false, false, false, false, SSize.Tiny)
+        case w if w < LargePhoneCut => (false, true, false, false, false, SSize.Small)
+        case _                      => (displayOffsets(p), true, true, true, true, SSize.Small)
       }
       val offsetColumn =
         p.offsetsDisplay match {
           case OffsetsDisplay.DisplayOffsets(x) =>
             Column(Column.props(ColWidths.OffsetWidthBase + x, "offset", label = "Offset", flexShrink = 0, disableSort = true, cellRenderer = stepStatusRenderer(p.offsetsDisplay))).some
           case _ => None
         }
-        List(
+
+        val hasDisperser = p.steps.exists(s => s.instrument === Instrument.GmosS || s.instrument === Instrument.GmosN)
+        (List(
           p.steps.map(i => Column(Column.props(ColWidths.ControlWidth, "ctl", label = "Icon", disableSort = true, cellRenderer = stepControlRenderer(i, p, recomputeRowHeightsCB), flexShrink = 0, className = SeqexecStyles.controlCellRow.htmlClass, headerRenderer = controlHeaderRenderer))),
           Column(Column.props(ColWidths.IdxWidth, "idx", label = "Step", disableSort = true, flexShrink = 0, cellRenderer = stepIdRenderer)).some,
           p.steps.map(i => Column(Column.props(ColWidths.StateWidth, "state", label = "Control", flexShrink = 1, flexGrow = 4, disableSort = true, cellRenderer = stepProgressRenderer(i, p)))),
           offsetColumn.filter(_ => offsetVisible),
-          p.steps.map(i => Column(Column.props(ColWidths.ExposureWidth, "exposure", label = "Exposure", flexShrink = 0, disableSort = true, className = SeqexecStyles.centeredCell.htmlClass, cellRenderer = stepExposureRenderer(i.instrument)))).filter(_ => exposureVisible),
+          p.steps.map(i => Column(Column.props(ColWidths.ExposureWidth, "exposure", label = "Exposure", flexShrink = 0, disableSort = true, className = SeqexecStyles.centeredCell.htmlClass, cellRenderer = stepExposureRenderer(i.instrument)))).filter(_ => exposureVisible)
+        ) ++
+          (if (hasDisperser) List(
```

Comment:
There has to be a much, much better way of doing this. @cquiroz, I think we should discuss this and see what you think the best plan is.